src/nvim/normal.c

```diff
@@ -4437,11 +4437,8 @@ static void nv_ident(cmdarg_T *cap)
       // Start insert mode in terminal buffer
       restart_edit = 'i';
 
-      add_map((char_u *)"<buffer> <esc> <Cmd>call jobstop(&channel)<CR>", TERM_FOCUS, true);
-      do_cmdline_cmd("autocmd TermClose <buffer> "
-                     " if !v:event.status |"
-                     "   exec 'bdelete! ' .. expand('<abuf>') |"
-                     " endif");
+      add_map((char_u *)"<buffer> <esc> <Cmd>call jobstop(&channel) | "
+              "exec 'bdelete! ' .. expand('<abuf>')<CR>", TERM_FOCUS, true);
```

Comment:
`<abuf>` won't be defined here since it's not executing in an autocmd context. Also, do you need to use `<bar>` instead of `|` (you certainly do when defining mappings in Vim script, I'm not sure if you do when calling `add_map`).

You may also want to wait to delete the buffer until the job has finished (i.e. after calling `jobwait()`).

Reply:
The buffer that needs to be deleted is always the current buffer, so the `<abuf>` is indeed not needed.

Reply:
Updated to remove `expand('<abuf>')` (3b28d63494b2240b2c11d71c0a399cdf42810afc). I guess this was working in my initial testing since it was expanding to the empty string.

Both `|` and `<bar>` work in my tests. I'll use `<bar>` to hopefully avoid the same question by a future reader of the code (de06fa252efb54907c5a8b85678443a680627e5b).

Thanks!